templates/skeleton/app/components/Aside.tsx

```diff
@@ -29,6 +35,23 @@ export function Aside({
   const {type: activeType, close} = useAside();
   const expanded = type === activeType;
 
+  useEffect(() => {
+    const abortController = new AbortController();
+
+    if (expanded) {
+      document.addEventListener(
+        'keypress',
+        function handler(event: KeyboardEvent) {
+          if (event.key === 'Escape') {
+            close();
+          }
+        },
+        {signal: abortController.signal},
+      );
+    }
+    return () => abortController.abort();
```

Comment:
Could you clean up the event listener here too? (in the case someone closes the cart without pressing Escape)

Reply:
The event listener is being cleaned up here actually, this is a relatively new technique using an abort controller instead of `removeEventListener`. Calling `abort` causes the event listener to be removed, which is passed as a [signal](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#signal) to the options.

It has [pretty good support across the board](https://caniuse.com/mdn-api_eventtarget_addeventlistener_options_parameter_options_signal_parameter), but can switch to a good old `removeEventListener` if desired.